### PR TITLE
Fix row/array/map type rendering

### LINKED ISSUE #2561: Rendering of ROW type is broken
The rendering of the ROW type in describe includes the words "Optional" and also doesn't render correctly when nested in an array

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
Fixes #2561

### CODE REVIEW

**Comment by electrum** [comment]
Does this need to be fixed for `MapType` too?

**Comment by nezihyigitbasi** [comment]
@electrum It seems so. Should I open a new PR for that or just update this?

**Comment by cberner** [comment]
I would just update this one

**Comment by nezihyigitbasi** [comment]
OK, updated the PR to fix the map type rendering too.

File: presto-main/src/main/java/com/facebook/presto/type/RowType.java [hunk 1]
```diff
@@ -66,18 +66,10 @@ public RowType(List<Type> fieldTypes, Optional<List<String>> fieldNames)
...
-            if (field.getName().isPresent()) {
-                fields.add(field.getName() + " " + field.getType().getDisplayName());
-            }
-            else {
-                fields.add(field.getType().getDisplayName());
-            }
-        }
-        return "ROW(" + Joiner.on(", ").join(fields) + ")";
+        List<String> fields = this.fields.stream().
+                map(field -> field.getType().getDisplayName()).
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/type/RowType.java]: You lost the `field.getName()` when you refactored this

File: presto-main/src/main/java/com/facebook/presto/type/RowType.java [hunk 2]
```diff
@@ -66,18 +66,17 @@ public RowType(List<Type> fieldTypes, Optional<List<String>> fieldNames)
     @Override
     public String getDisplayName()
     {
```
Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/type/RowType.java]: Updated row type rendering to be similar to Hive.

Here is what Hive shows:
```
array<struct<PRODUCT_ID:bigint,TIMESTAMPS:string>>
```

Here is what Presto shows:
```
array<row<product_id:bigint, timestamps:varchar>>
```

**Comment by nezihyigitbasi** [comment]
@cberner thanks, updated.

**Comment by cberner** [comment]
@nezihyigitbasi the SQL standard defines a format for row types (https://mariadb.com/kb/en/sql-99/row-data-types/). So we should probably stick to that, at least for now

File: presto-main/src/main/java/com/facebook/presto/type/RowType.java [hunk 3]
```diff
@@ -67,17 +67,17 @@ public RowType(List<Type> fieldTypes, Optional<List<String>> fieldNames)
...
         // Convert to standard sql name
-        List<String> fields = new ArrayList<>();
-        for (int i = 0; i < this.fields.size(); i++) {
-            RowField field = this.fields.get(i);
+        List<String> fieldDisplayNames = new ArrayList<>();
```
Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/type/RowType.java]: @cberner Updated to conform to the standard. I made the type name all lowercase to be consistent across all complex types, do we want to do that?

```
array<row(product_id bigint, timestamps varchar)>
```

Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/type/RowType.java]: Ya, that seems fine to me

**Comment by cberner** [comment]
Merged. Thanks!

**Pull request closed without merging**